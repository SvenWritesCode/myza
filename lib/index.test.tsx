import React from 'react'
import styled from 'styled-components'
import Myza, { Components } from './'

const SimpleWrapper = styled.div`
  background: pink;
`

const SimpleEmail = ({ firstName }) => <SimpleWrapper>Welcome to Curri, {firstName}</SimpleWrapper>

it('renders correctly', () => {
  const rendered = Myza.renderEmail(SimpleEmail, { firstName: 'Brian' })
  const tree = JSON.stringify({ body: rendered })
  expect(tree).toMatchSnapshot();
});


it('renders with custom font-family', () => {
  const rendered = Myza.renderEmail(SimpleEmail, { firstName: 'Brian' }, { fontFamily: 'Comic Sans' })
  const tree = JSON.stringify({ body: rendered })
  expect(tree).toMatchSnapshot();
});

it('renders with Myza component', () => {
  const CenteredEmail = ({ firstName }) => <Components.Center>Welcome to Curri, {firstName}</Components.Center>
  const rendered = Myza.renderEmail(CenteredEmail, { firstName: 'Brian' })
  const tree = JSON.stringify({ body: rendered })
  expect(tree).toMatchSnapshot();
});
