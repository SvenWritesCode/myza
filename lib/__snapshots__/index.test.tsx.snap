// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"{\\"body\\":\\"<!doctype html>\\\\n<html xmlns=\\\\\\"http://www.w3.org/1999/xhtml\\\\\\" xmlns:v=\\\\\\"urn:schemas-microsoft-com:vml\\\\\\" xmlns:o=\\\\\\"urn:schemas-microsoft-com:office:office\\\\\\">\\\\n\\\\n  <head>\\\\n    <title>\\\\n\\\\n    </title>\\\\n    <!--[if !mso]><!-- -->\\\\n    <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n    <!--<![endif]-->\\\\n    <meta http-equiv=\\\\\\"Content-Type\\\\\\" content=\\\\\\"text/html; charset=UTF-8\\\\\\">\\\\n    <meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width, initial-scale=1\\\\\\">\\\\n\\\\n    <!--[if !mso]><!-->\\\\n    <style type=\\\\\\"text/css\\\\\\">\\\\n      @media only screen and (max-width:480px) {\\\\n        @-ms-viewport {\\\\n          width: 320px;\\\\n        }\\\\n\\\\n        @viewport {\\\\n          width: 320px;\\\\n        }\\\\n      }\\\\n    </style>\\\\n    <!--<![endif]-->\\\\n\\\\n    <!--[if mso]>\\\\n        <xml>\\\\n        <o:OfficeDocumentSettings>\\\\n          <o:AllowPNG/>\\\\n          <o:PixelsPerInch>96</o:PixelsPerInch>\\\\n        </o:OfficeDocumentSettings>\\\\n        </xml>\\\\n        <![endif]-->\\\\n    <!--[if lte mso 11]>\\\\n        <style type=\\\\\\"text/css\\\\\\">\\\\n          .outlook-group-fix { width:100% !important; }\\\\n        </style>\\\\n        <![endif]-->\\\\n\\\\n  </head>\\\\n\\\\n  <body style=\\\\\\"margin: 0; padding: 0; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; font-family: Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';\\\\\\">\\\\n\\\\n\\\\n    <div class=\\\\\\"sc-AykKC\\\\\\" style=\\\\\\"background: pink;\\\\\\">Welcome to Curri, Brian</div>\\\\n  </body>\\\\n\\\\n</html>\\"}"`;

exports[`renders with Myza component 1`] = `"{\\"body\\":\\"<!doctype html>\\\\n<html xmlns=\\\\\\"http://www.w3.org/1999/xhtml\\\\\\" xmlns:v=\\\\\\"urn:schemas-microsoft-com:vml\\\\\\" xmlns:o=\\\\\\"urn:schemas-microsoft-com:office:office\\\\\\">\\\\n\\\\n  <head>\\\\n    <title>\\\\n\\\\n    </title>\\\\n    <!--[if !mso]><!-- -->\\\\n    <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n    <!--<![endif]-->\\\\n    <meta http-equiv=\\\\\\"Content-Type\\\\\\" content=\\\\\\"text/html; charset=UTF-8\\\\\\">\\\\n    <meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width, initial-scale=1\\\\\\">\\\\n\\\\n    <!--[if !mso]><!-->\\\\n    <style type=\\\\\\"text/css\\\\\\">\\\\n      @media only screen and (max-width:480px) {\\\\n        @-ms-viewport {\\\\n          width: 320px;\\\\n        }\\\\n\\\\n        @viewport {\\\\n          width: 320px;\\\\n        }\\\\n      }\\\\n    </style>\\\\n    <!--<![endif]-->\\\\n\\\\n    <!--[if mso]>\\\\n        <xml>\\\\n        <o:OfficeDocumentSettings>\\\\n          <o:AllowPNG/>\\\\n          <o:PixelsPerInch>96</o:PixelsPerInch>\\\\n        </o:OfficeDocumentSettings>\\\\n        </xml>\\\\n        <![endif]-->\\\\n    <!--[if lte mso 11]>\\\\n        <style type=\\\\\\"text/css\\\\\\">\\\\n          .outlook-group-fix { width:100% !important; }\\\\n        </style>\\\\n        <![endif]-->\\\\n\\\\n  </head>\\\\n\\\\n  <body style=\\\\\\"margin: 0; padding: 0; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; font-family: Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';\\\\\\">\\\\n\\\\n    <style data-styled data-styled-version=\\\\\\"5.0.0-5.canary-sheet\\\\\\"></style>\\\\n    <table width=\\\\\\"100%\\\\\\" border=\\\\\\"0\\\\\\" cellspacing=\\\\\\"0\\\\\\" cellpadding=\\\\\\"0\\\\\\" style=\\\\\\"border-collapse: collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt;\\\\\\">\\\\n      <tr>\\\\n        <td align=\\\\\\"center\\\\\\" style=\\\\\\"border-collapse: collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt;\\\\\\">Welcome to Curri, Brian</td>\\\\n      </tr>\\\\n    </table>\\\\n  </body>\\\\n\\\\n</html>\\"}"`;

exports[`renders with custom font-family 1`] = `"{\\"body\\":\\"<!doctype html>\\\\n<html xmlns=\\\\\\"http://www.w3.org/1999/xhtml\\\\\\" xmlns:v=\\\\\\"urn:schemas-microsoft-com:vml\\\\\\" xmlns:o=\\\\\\"urn:schemas-microsoft-com:office:office\\\\\\">\\\\n\\\\n  <head>\\\\n    <title>\\\\n\\\\n    </title>\\\\n    <!--[if !mso]><!-- -->\\\\n    <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n    <!--<![endif]-->\\\\n    <meta http-equiv=\\\\\\"Content-Type\\\\\\" content=\\\\\\"text/html; charset=UTF-8\\\\\\">\\\\n    <meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width, initial-scale=1\\\\\\">\\\\n\\\\n    <!--[if !mso]><!-->\\\\n    <style type=\\\\\\"text/css\\\\\\">\\\\n      @media only screen and (max-width:480px) {\\\\n        @-ms-viewport {\\\\n          width: 320px;\\\\n        }\\\\n\\\\n        @viewport {\\\\n          width: 320px;\\\\n        }\\\\n      }\\\\n    </style>\\\\n    <!--<![endif]-->\\\\n\\\\n    <!--[if mso]>\\\\n        <xml>\\\\n        <o:OfficeDocumentSettings>\\\\n          <o:AllowPNG/>\\\\n          <o:PixelsPerInch>96</o:PixelsPerInch>\\\\n        </o:OfficeDocumentSettings>\\\\n        </xml>\\\\n        <![endif]-->\\\\n    <!--[if lte mso 11]>\\\\n        <style type=\\\\\\"text/css\\\\\\">\\\\n          .outlook-group-fix { width:100% !important; }\\\\n        </style>\\\\n        <![endif]-->\\\\n\\\\n  </head>\\\\n\\\\n  <body style=\\\\\\"margin: 0; padding: 0; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; font-family: Comic Sans;\\\\\\">\\\\n\\\\n\\\\n    <div class=\\\\\\"sc-AykKC\\\\\\" style=\\\\\\"background: pink;\\\\\\">Welcome to Curri, Brian</div>\\\\n  </body>\\\\n\\\\n</html>\\"}"`;
